presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/util/Timestamps.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.iceberg.util;
+
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.type.LongTimestampWithTimeZone;
+import io.prestosql.spi.type.TimeZoneKey;
+import io.prestosql.spi.type.Type;
+
+import static io.prestosql.spi.type.TimestampType.createTimestampType;
+import static io.prestosql.spi.type.TimestampWithTimeZoneType.createTimestampWithTimeZoneType;
+import static io.prestosql.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+import static java.lang.Math.toIntExact;
+
+public final class Timestamps
+{
+    public static final Type TIMESTAMP_MICROS = createTimestampType(6);
+    public static final Type TIMESTAMP_TZ_MICROS = createTimestampWithTimeZoneType(6);
+
+    private Timestamps() {}
+
+    public static long toMicros(LongTimestampWithTimeZone timestamp)
+    {
+        return (timestamp.getEpochMillis() * 1000) + (timestamp.getPicosOfMilli() / PICOSECONDS_PER_MICROSECOND);
```

Comment:
I'd add MICROSECONDS_PER_MILLISECOND to Timestamps and use it here.

Also, `(timestamp.getPicosOfMilli() / PICOSECONDS_PER_MICROSECOND` should round.

Reply:
This method is only used for Iceberg timestamps which have microsecond precision, so rounding is not a concern, but I'll change it since there's no downside to doing the calculation correctly.